### PR TITLE
Add Introductory Price to InAppBillingProduct

### LINKED ISSUE #123: Add Introductory Price to InAppBillingProduct
In android, and iOS as well, fields about the Introductory price would be very useful. 
iOS Documentation about introductory price: https://developer.apple.com/documentation/storekit/in_app_purchase/offering_introductory_prices_in_your_app
Android Documentation about introductory price: https://developer.android.com/google/play/billing/billing_subscriptions.html#intro

## Bug Information
No Fields available regarding Introductory price

Version Number of Plugin: 1.2.4 (and the latest github version)
Device Tested On: iPhone 6/ Samsung S5
Simulator Tested On: N/A
Version of VS: 7.3.3
Version of Xamarin:  Xamarin.iOS -Version: 11.6.1.4 (Visual Studio Community)
Xamarin.Android Version: 8.1.5.0 (Visual Studio Community)
Versions of other things you are using: N/A

### Steps to reproduce the Behavior
Call GetProductInfoAsync.

### Expected Behavior
InAppBillingProduct containing fields regarding wether Introductory price is available and the specific introductory prices for the product. 

### Actual Behavior
No fields available for introductory price in InAppBillingProduct after calling GetProductInfoAsync.

### Code snippet
N/A

### Screenshots
N/A

### PR DESCRIPTION
**Pull request by PhilippeCreytens**
Please take a moment to fill out the following:

Fixes #123  .

Changes Proposed in this pull request:
- Add introductory price to Android
- Add introductory price to iOS
- Implement boolean to check if an introductory price is available

### CODE REVIEW

**Comment by PhilippeCreytens** [comment]
Hey James,
I have updated the pull request to only add Introductory price since iOS 11.2.
Regarding the android, yes this is available in the binding you have made, if an introductory price is available then this is part of the json that is being processed. An introductory price is available from the Google Play Store version 3.5, the earliest version now available that supports subscriptions.

**Comment by marza91** [comment]
Going to test this for our app, as we need this before we can do a release. Is there no way around needing iOS 11.2 to get these prices?
We are supposed to support iOS 8, but might have to reconsider if we can't get the prices to show up correctly.

**Comment by jamesmontemagno** [comment]
@marza91 they added the api in 11.2 so it is impossible to have it before any release before that. 

There really isn't a reason to support anything older than iOS 10: https://data.apteligent.com/ios/

File: src/Plugin.InAppBilling.iOS/InAppBillingImplementation.cs [hunk 1]
```diff
@@ -56,15 +56,18 @@ public InAppBillingImplementation()
...
+
+            return products.Select(p => new InAppBillingProduct {
+                LocalizedPrice = p.LocalizedPrice(),
+                MicrosPrice = (long)(p.Price.DoubleValue * 1000000d),
+                Name = p.LocalizedTitle,
+                ProductId = p.ProductIdentifier,
+                Description = p.LocalizedDescription,
+                CurrencyCode = p.PriceLocale?.CurrencyCode ?? string.Empty,
+                LocalizedIntroductoryPrice = operatingSystemHasIntroductoryPrice && p.IntroductoryPrice != null ? p.IntroductoryPrice.LocalizedPrice() : "",
+                MicrosIntroductoryPrice = operatingSystemHasIntroductoryPrice && p.IntroductoryPrice != null ? (long)(p.Price.DoubleValue * 1000000d) : 0
```
Comment (jamesmontemagno) [inline src/Plugin.InAppBilling.iOS/InAppBillingImplementation.cs]: shouldn't this be looking at p.IntroductoryPrice.DoubleValue?

Comment (jamesmontemagno) [reply src/Plugin.InAppBilling.iOS/InAppBillingImplementation.cs]: maybe we should refactor this into a common method?

File: src/Plugin.InAppBilling.iOS/InAppBillingImplementation.cs [hunk 2]
```diff
@@ -56,15 +56,18 @@ public InAppBillingImplementation()
...
-			return products.Select(p => new InAppBillingProduct
-			{
-				LocalizedPrice = p.LocalizedPrice(),
-				MicrosPrice = (long)(p.Price.DoubleValue * 1000000d),
-				Name = p.LocalizedTitle,
-				ProductId = p.ProductIdentifier,
-				Description = p.LocalizedDescription,
-				CurrencyCode = p.PriceLocale?.CurrencyCode ?? string.Empty
-			});
+            var operatingSystemHasIntroductoryPrice = NSProcessInfo.ProcessInfo.IsOperatingSystemAtLeastVersion(new NSOperatingSystemVersion(11, 2, 0));
```
Comment (jamesmontemagno) [inline src/Plugin.InAppBilling.iOS/InAppBillingImplementation.cs]: This is wrong. Put it at the class level:

bool IsiOS112 => UIDevice.CurrentDevice.CheckSystemVersion (11,2);

**Comment by jamesmontemagno** [comment]
Added a few more required tweaks to this one

**Comment by PhilippeCreytens** [comment]
Hey James,

I have updated the PR, an extension method is now used to check whether an introduction price can be used and I have updated the DoubleValue to target the correct value.

**Comment by jamesmontemagno** [comment]
Any changes or anything needed for UWP?

**Comment by PhilippeCreytens** [comment]
No changes are needed for UWP, the InAppBillingProduct will say no IntroductoryPrice is available, according to the microsoft documentation I do not see any reference to an introductory price, only if the item is on sale or not but this is not the same as an introductory price.

**Pull request merged**